devlib/instrument/arm_energy_probe.py

```diff
@@ -0,0 +1,130 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# pylint: disable=W0613,E1101,access-member-before-definition,attribute-defined-outside-init
+from __future__ import division
+import os
+import csv
+import subprocess
+import signal
+import struct
+import sys
+
+import tempfile
+import shutil
+
+from devlib.instrument import Instrument, CONTINUOUS, MeasurementsCsv
+from devlib.exception import HostError
+from devlib.utils.misc import which
+
+sys.path.append(os.path.dirname(sys.modules[__name__].__file__))
+from parse_aep import AEP_parser
+
+class ArmEnergyProbeInstrument(Instrument):
+
+    mode = CONTINUOUS
+
+    name = 'arm_energy_probe'
+    description = """Collects power traces using the ARM Energy Probe.
+
+                     This instrument requires ``arm-probe`` utility to be installed in the workload automation
+                     host and be in the PATH. arm-probe is available here ``https://git.linaro.org/tools/arm-probe.git`` .
+                     ARM energy probe device can simultaneously collect power from up to 3 power rails and
+                     arm-probe utility can record data from several devices simultaneously.
+
+                     To connect the energy probe on a rail, connect the white wire to the pin that is closer to the
+                     Voltage source and the black wire to the pin that is closer to the load (the SoC or the device
+                     you are probing). Between the pins there should be a shunt resistor of known resistance in the
+                     range of 5 to 500 mOhm but the voltage on the shunt resistor must stay smaller than 165mV.
+                     The resistance of the shunt resistors is a mandatory parameter to be set in the ``config`` file.
+                    """
+
+    MAX_CHANNELS = 12 # 4 Arm Energy Probes
+
+    def __init__(self, target, config_file='./config-aep', ):
+        super(ArmEnergyProbeInstrument, self).__init__(target)
+        self.arm_probe = which('arm-probe')
+        if self.arm_probe is None:
+            raise HostError('arm-probe must be installed on the host')
+        #todo detect is config file exist
+        self.attributes = ['power', 'voltage', 'current']
+        self.sample_rate_hz = 10000
+        self.config_file = config_file
+
+        self.parser = AEP_parser()
+        #TODO make it generic
+        topo = self.parser.topology_from_config(self.config_file)
+        for item in topo:
+            if item == 'time':
+                self.add_channel('timestamp', 'time')
+                continue
```

Comment:
no it has to be a continue but it can be an else as well.
might be clearer with else ?
+            if item == 'time':
+                self.add_channel('timestamp', 'time')
￼+            else:
+                 self.add_channel(item, 'power')

Reply:
oh, right, I see. yes, with an `else` it would be clearer.